recipe/patches/0032-loosen-hermetic-python.patch

```diff
@@ -0,0 +1,185 @@
+From 7117a31a1b8dda2e0313b276a5697a5ce09fcff2 Mon Sep 17 00:00:00 2001
+From: Charles Bousseau <cbousseau@anaconda.com>
+Date: Tue, 3 Sep 2024 09:01:50 -0400
+Subject: [PATCH] loosen hermetic python
+
+Tensorflow registers python dependencies using rules_python in order
+to be hermetic to system installed python packages.
+Even if packages are installed, packages are downloaded from pypi at 
+build time.
+On linux-aarch64, a number of requested packages are not available.
+This removes some declared dependencies allowing the build to proceed.
```

Comment:
I've been looking into this hermetic python business and haven't understood what's going on properly but was wondering if you knew the following?
- if "hermetic python" is switched on, why doesn't that mean that the tf build system will download its own python and python dependencies rather than using ours? Do we want to switch hermetic python off?
- how do the bazel files provided in this feedstock interact with the bazel files in the tf build system? Are they copied over similar files in the tree or something like that?
Basically I'm curious about why removing these pypi downloads is inconsequential. I.e. how the tf build system knows to get our dependencies anyway.

Reply:
Hermetic python is not a flag you can switch off unfortunately. Patch 0011 ensures we are using our python interpreter (which means whatever package installed in this environment is available as well). 
The tf build system does not know how about this however. Tensorflow uses https://github.com/bazelbuild/rules_python targets to configure pypi dependencies at build time (for example, `@pypi_keras//:pkg`  essentially triggers a `pip install`). A proper way would be to enhance that project to handle system dependencies instead. This is a big undertaking. 
The python dependencies are not vendored though (you can verify this by extracting the resulting tensorflow package). So at runtime, our dependencies are used as expected. 

The bazel files in the feedstock do not replace upstream files. Rather, the extra configuration is added through `.bazelrc` set in our build scripts.

Reply:
> at runtime, our dependencies are used as expected

and, indeed at build time?
Thanks sounds good to me.